include/gsl/gsl_byte

```diff
@@ -37,10 +37,30 @@
 #pragma push_macro("noexcept")
 #define noexcept /*noexcept*/
 #endif           // _MSC_VER <= 1800
+
+#if _HAS_STD_BYTE 
+
```

Comment:
@neilmacintosh Sure.  I generally prefer and encourage testing for compiler version than feature test macros.  Compiler versions tend to be more reliable :-)